tests/models/test_token_classification_with_seq2seq_encoder_and_crf.py

```diff
@@ -0,0 +1,258 @@
+from transformers.modeling_outputs import BaseModelOutputWithPoolingAndCrossAttentions
+
+from pie_models.models import TokenClassificationModelWithSeq2SeqEncoderAndCrf
+from tests import FIXTURES_ROOT
+
+DUMP_FIXTURES = False
+FIXTURES_TASKMODULE_DATA_PATH = (
+    FIXTURES_ROOT / "taskmodules" / "TransformerTokenClassificationTaskModule"
+)
+
+
+@pytest.fixture(scope="module")
+def documents(dataset):
+    return dataset["train"]
+
+
+@pytest.mark.skipif(
+    condition=not DUMP_FIXTURES, reason="Only need to dump the data if taskmodule has changed"
+)
+def test_dump_fixtures(documents):
+    tokenizer_name_or_path = "bert-base-cased"
+    taskmodule = TransformerTokenClassificationTaskModule(
+        tokenizer_name_or_path=tokenizer_name_or_path,
+    )
+    taskmodule.prepare(documents)
+    encodings = taskmodule.encode(documents, encode_target=True, as_dataset=True)
+    batch_encoding = taskmodule.collate(encodings[:4])
+
+    FIXTURES_TASKMODULE_DATA_PATH.mkdir(parents=True, exist_ok=True)
+    filepath = FIXTURES_TASKMODULE_DATA_PATH / "batch_encoding_inputs.json"
+
+    inputs = {key: tensor.tolist() for key, tensor in batch_encoding[0].items()}
+    targets = batch_encoding[1].tolist()
+    converted_batch_encoding = {
+        "inputs": inputs,
+        "targets": targets,
+    }
+    with open(filepath, "w") as f:
+        json.dump(converted_batch_encoding, f)
+    return converted_batch_encoding
+
+
+@pytest.fixture
+def batch(documents):
+    filepath = FIXTURES_TASKMODULE_DATA_PATH / "batch_encoding_inputs.json"
+    with open(filepath) as f:
+        batch_encoding = json.load(f)
+    inputs = {
+        "input_ids": torch.LongTensor(batch_encoding["inputs"]["input_ids"]),
+        "token_type_ids": torch.LongTensor(batch_encoding["inputs"]["token_type_ids"]),
+        "attention_mask": torch.LongTensor(batch_encoding["inputs"]["attention_mask"]),
+    }
+    targets = torch.LongTensor(batch_encoding["targets"])
+    return (inputs, targets)
+
+
+def get_model(
+    monkeypatch,
+    model_type,
+    num_classes,
+    batch_size,
+    seq_len,
+    add_dummy_linear=False,
+    **model_kwargs,
+):
+    class MockConfig:
+        def __init__(
+            self,
+            hidden_size: int = 10,
+            classifier_dropout: float = 0.1,
+            model_type=model_type,
+        ) -> None:
+            self.hidden_size = hidden_size
+            self.model_type = model_type
+            if self.model_type == "bert":
+                self.hidden_dropout_prob = classifier_dropout
+            elif self.model_type == "albert":
+                self.classifier_dropout_prob = classifier_dropout
+            elif self.model_type == "distilbert":
+                self.seq_classif_dropout = classifier_dropout
+            elif self.model_type == "longformer":
+                self.hidden_dropout_prob = classifier_dropout
+            else:
+                self.classifier_dropout = classifier_dropout
```

Comment:
Can you not use `HF_MODEL_TYPE_TO_CLASSIFIER_DROPOUT_ATTRIBUTE` here? e.g. with `setattr`